### ISSUE TITLE
Y Axis ticks not display correctly

### ISSUE THREAD

**Issue by idanlevi1** [FormidableLabs/victory-native-xl#189]
I want to display in Y axis 3 ticks with equal length between each other
I use these props:
`tickCount: { x: 1, y: 3 },`
and sometimes display 3 ticks and sometimes 2 ticks, the position is not consistently 
![image](https://github.com/FormidableLabs/victory-native-xl/assets/21155847/cb63e33d-b3bf-4ddb-ac7f-b54b79ea5c4f)
![image](https://github.com/FormidableLabs/victory-native-xl/assets/21155847/bc2af9fe-d28f-4628-8e08-41334668714a)

I need to display 3 ticks in every case and constant positions

**Comment by kyle-belle** [comment]
I get this on the x axis as well. It also sometime just cuts off or goes out of bounds and doesn't space them out evenly.  I was also wondering if we could tilt the x axis labels

In all cases below i specified the tick count should be 4 for the x axis
```tsx
<CartesianChart
  data={data}
  xKey="x"
  yKeys={['y']}
  // chartPressState={chartPressState}
  domain={{
    x: [
      firstData?.date.getTime() || dateRange[0].getTime(),
      lastIntake !== firstData?lastData?.date.getTime() || dateRange[1].getTime() : dateRange[1].getTime(),
    ],
   y: [0],
  }}
  domainPadding={{left: 0, right: 0, top: 20}}
  axisOptions={{
    font: graphFont,
    tickCount: {x: 4, y: 5},
    lineWidth: 2,
    formatXLabel: timeMs => {
      return dayjs(timeMs).format('YYYY/MM/DD');
    },
  }}
  // domainPadding={20}
  /* theme={VictoryTheme.material} */
/>
```

![Screenshot_20240205-190127](https://github.com/FormidableLabs/victory-native-xl/assets/43228761/a40dbc83-6b8f-47d4-a86d-69cfb6d51a0a)
![Screenshot_20240205-190148](https://github.com/FormidableLabs/victory-native-xl/assets/43228761/24727f20-d671-436e-b283-d867e98522dc)
![Screenshot_20240205-190158](https://github.com/FormidableLabs/victory-native-xl/assets/43228761/ec92cc25-7dba-441b-b09f-c5893879f5c9)
![Screenshot_20240205-190305](https://github.com/FormidableLabs/victory-native-xl/assets/43228761/2c3fef20-516e-4975-8340-1d0a00870d63)

**Comment by masiddee** [comment]
> I want to display in Y axis 3 ticks with equal length between each other I use these props: `tickCount: { x: 1, y: 3 },` and sometimes display 3 ticks and sometimes 2 ticks, the position is not consistently ![image](https://private-user-images.githubusercontent.com/21155847/302109459-cb63e33d-b3bf-4ddb-ac7f-b54b79ea5c4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNDg3NjUsIm5iZiI6MTcwOTE0ODQ2NSwicGF0aCI6Ii8yMTE1NTg0Ny8zMDIxMDk0NTktY2I2M2UzM2QtYjNiZi00ZGRiLWFjN2YtYjU0Yjc5ZWE1YzRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDE5Mjc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY1ODIxNjNhMDQ2YzQ5NThiZDhkNzViZWJjYzc1MzI1MjY5MTM2YWMzOWFjNzAzOTNiOTJiNjA4MGQwZTc2MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kzA-z8R_UtSbazJqp-tdLGfSsCC3aFF-35CSw4IXvNE) ![image](https://private-user-images.githubusercontent.com/21155847/302109475-bc2af9fe-d28f-4628-8e08-41334668714a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkxNDg3NjUsIm5iZiI6MTcwOTE0ODQ2NSwicGF0aCI6Ii8yMTE1NTg0Ny8zMDIxMDk0NzUtYmMyYWY5ZmUtZDI4Zi00NjI4LThlMDgtNDEzMzQ2Njg3MTRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI4VDE5Mjc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRjYTNmMTIyMjk4N2RiMjBmYjE4NDg4NGVhMmVjODAwNTNlYTAxNzAwNTZmMTI4MDI4YzFkMzI5M2FiMzA2YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HmqpeWxm_9IbgFdvYuoxFqOVqng_6IaSL80pgJcnIPE)
> 
> I need to display 3 ticks in every case and constant positions

Thanks for sharing this with us @idanlevi1! I took a deep-dive into the logic that we use to derive the `tickCount` here, and ultimately found that the behavior you are seeing is behaving as expected.

Essentially, under the hood, VNXL uses [d3-scale.ticks](https://observablehq.com/@d3/scale-ticks?collection=@d3/d3-scale) to [generate the x/y axis ticks](https://github.com/FormidableLabs/victory-native-xl/blob/main/lib/src/cartesian/components/CartesianAxis.tsx#L93) that are rendered on a given chart. And here's what this function signature states with regards to the ticks returned:

```js
 /**
     * Returns approximately count representative values from the scale’s domain.
     *
     * If count is not specified, it defaults to 10.
     *
     * The returned tick values are uniformly spaced, have human-readable values (such as multiples of powers of 10),
     * and are guaranteed to be within the extent of the domain. Ticks are often used to display reference lines, or tick marks, in conjunction with the visualized data.
     * The specified count is only a hint; the scale may return more or fewer values depending on the domain. See also d3-array’s ticks.
     *
     * @param count Optional approximate number of ticks to be returned. If count is not specified, it defaults to 10.
     */
    ticks(count?: number): number[];
```

TL;DR -- the `tickCount` value is not a hard and fast constraint, but instead is effectively an approximation that takes into account other factors such as domain, label width, etc. to display the optimal number of axis.

And with regards to your post, @kyle-belle, considering the above, it appears the varying width of the xLabels, as well as the derived domain values in your code may be impacting how the ticks are rendering in your examples.

With that being said, I plan to spin up a PR today that would update our docs to better reflect this reality when using the `tickCount` property. I'll leave this issue open until that PR is merged.

I hope this helps!

**Comment by masiddee** [comment]
PR to update our documentation was merged. Closing this issue.
